### PR TITLE
MafUtil cleanup, Added imported MAF cols documentation

### PR DESCRIPTION
**Pull request by ao508**
# What? Why?
Clean up unused columns from [MafUtil](https://github.com/cBioPortal/cbioportal/blob/master/core/src/main/java/org/mskcc/cbio/maf/MafUtil.java) and related code in [ExtendedMutationutil](https://github.com/cBioPortal/cbioportal/blob/master/core/src/main/java/org/mskcc/cbio/portal/util/ExtendedMutationUtil.java) and [MafRecord](https://github.com/cBioPortal/cbioportal/blob/master/core/src/main/java/org/mskcc/cbio/maf/MafRecord.java). 

Changes proposed in this pull request:
- Remove columns from `MafUtil` that aren't used for resolving fields that are imported into `mutation` and `mutation_event`

**Open up discussion for what to do with MAF fields which map directly to database fields but are ignored during import / not used to set corresponding fields when constructing database objects for import**
```
- ONCOTATOR_REFSEQ_MRNA_ID (`"ONCOTATOR_REFSEQ_MRNA_ID"`) --> `mutation_event.oncotator_refseq_mrna_id`
- ONCOTATOR_UNIPROT_ENTRY_NAME (`"ONCOTATOR_UNIPROT_ENTRY_NAME"`) --> `mutation_event.oncotator_uniprot_entry_name`
- ONCOTATOR_UNIPROT_ACCESSION (`"ONCOTATOR_UNIPROT_ACCESSION"`) --> `mutation_event.oncotator_uniprot_accession`
- ONCOTATOR_CODON_CHANGE (`"ONCOTATOR_CODON_CHANGE"`) --> `mutation_event.oncotator_codon_change`
- ONCOTATOR_PROTEIN_POS_START (`"ONCOTATOR_PROTEIN_POS_START"`) --> `mutation_event.oncotator_protein_pos_start`
- ONCOTATOR_PROTEIN_POS_END (`"ONCOTATOR_PROTEIN_POS_END"`) --> `mutation_event.oncotator_protein_pos_end`
```
**NOTE: I highly recommend viewing diff w/o whitespace changes**

Columns proposed for removal (MafUtil field name (`"MAF_COLUMN_NAME"`)):
- TRANSCRIPT (`"Transcript"`)
- ONCOTATOR_COSMIC_OVERLAPPING (`"ONCOTATOR_COSMIC_OVERLAPPING"`)
- ONCOTATOR_DBSNP_VAL_STATUS (`"ONCOTATOR_DBSNP_VAL_STATUS"`)
- ONCOTATOR_PROTEIN_CHANGE (`"ONCOTATOR_PROTEIN_CHANGE"`)
- ONCOTATOR_GENE_SYMBOL (`"ONCOTATOR_GENE_SYMBOL"`)
- ONCOTATOR_REFSEQ_PROT_ID (`"ONCOTATOR_REFSEQ_PROT_ID"`)
- ONCOTATOR_TRANSCRIPT_CHANGE (`"ONCOTATOR_TRANSCRIPT_CHANGE"`)
- ONCOTATOR_EXON_AFFECTED (`"ONCOTATOR_EXON_AFFECTED"`)
- ONCOTATOR_PROTEIN_CHANGE_BE (`"ONCOTATOR_PROTEIN_CHANGE_BEST_EFFECT"`)
- ONCOTATOR_VARIANT_CLASSIFICATION_BE (`"ONCOTATOR_VARIANT_CLASSIFICATION_BEST_EFFECT"`)
- ONCOTATOR_GENE_SYMBOL_BE (`"ONCOTATOR_GENE_SYMBOL_BEST_EFFECT"`)
- ONCOTATOR_REFSEQ_MRNA_ID_BE (`"ONCOTATOR_REFSEQ_MRNA_ID_BEST_EFFECT"`)
- ONCOTATOR_REFSEQ_PROT_ID_BE (`"ONCOTATOR_REFSEQ_PROT_ID_BEST_EFFECT"`)
- ONCOTATOR_UNIPROT_ENTRY_NAME_BE (`"ONCOTATOR_UNIPROT_ENTRY_NAME_BEST_EFFECT"`)
- ONCOTATOR_UNIPROT_ACCESSION_BE (`"ONCOTATOR_UNIPROT_ACCESSION_BEST_EFFECT"`)
- ONCOTATOR_CODON_CHANGE_BE (`"ONCOTATOR_CODON_CHANGE_BEST_EFFECT"`)
- ONCOTATOR_TRANSCRIPT_CHANGE_BE (`"ONCOTATOR_TRANSCRIPT_CHANGE_BEST_EFFECT"`)
- ONCOTATOR_EXON_AFFECTED_BE (`"ONCOTATOR_EXON_AFFECTED_BEST_EFFECT"`)
- ONCOTATOR_PROTEIN_POS_START_BE (`"ONCOTATOR_PROTEIN_POS_START_BEST_EFFECT"`)
- ONCOTATOR_PROTEIN_POS_END_BE (`"ONCOTATOR_PROTEIN_POS_END_BEST_EFFECT"`)
- MA_PROTEIN_CHANGE (`"MA:protein.change"`)

# Checks
- [ ] Runs on Heroku.
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Google Style Guide](https://github.com/google/styleguide).
- [ ] If this is a feature, the PR is to rc. If this is a bug fix, the PR is to master.

Signed-off-by: Angelica Ochoa <aochoa4230@gmail.com>

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/portal/util/ExtendedMutationUtil.java [hunk 1]
```diff
@@ -47,319 +47,296 @@
...
-	 * If none of the above is valid then returns "MUTATED".
-	 *
-	 * @param parts     current mutation as split parts of the line
-	 * @param record    MAF record for the current line
-	 * @return          most accurate protein change
-	 */
-	public static String getProteinChange(String[] parts, MafRecord record)
-	{
-		// try annotator value first
-		//String proteinChange = record.getOncotatorProteinChange();
```
Comment (angelicaochoa) [inline core/src/main/java/org/mskcc/cbio/portal/util/ExtendedMutationUtil.java]: removed commented out reference to `record.getOncotatorProteinChange()`

File: core/src/main/java/org/mskcc/cbio/maf/MafRecord.java [hunk 2]
```diff
@@ -88,57 +87,38 @@
     private int normalDepth;
     private float normalVaf;
 
-	// custom annotator columns
```
Comment (angelicaochoa) [inline core/src/main/java/org/mskcc/cbio/maf/MafRecord.java]: cleaned up columns that are never referenced when constructing db objects for `mutation`, `mutation_event` imports

File: core/src/main/java/org/mskcc/cbio/maf/MafUtil.java [hunk 3]
```diff
@@ -45,121 +41,109 @@
...
-	public static final String SEQUENCING_PHASE = "Sequencing_Phase";
-	public static final String SEQUENCE_SOURCE = "Sequence_Source";
-	public static final String VALIDATION_METHOD = "Validation_Method";
-	public static final String SCORE = "Score";
-	public static final String BAM_FILE = "BAM_File";
-	public static final String SEQUENCER = "Sequencer";
-
-	// non-standard columns
-	public static final String AMINO_ACID_CHANGE = "Amino_Acid_Change";
-	public static final String TRANSCRIPT = "Transcript";
```
Comment (angelicaochoa) [inline core/src/main/java/org/mskcc/cbio/maf/MafUtil.java]: unused anywhere

File: core/src/main/java/org/mskcc/cbio/maf/MafUtil.java [hunk 4]
```diff
@@ -208,79 +192,57 @@
     private int normalVafIndex = -1;
 
     // default Oncotator column indices
```
Comment (angelicaochoa) [inline core/src/main/java/org/mskcc/cbio/maf/MafUtil.java]: * Removed several oncotator fields (+ a couple others) that are not referenced/used when constructing `mutation`, `mutation_event` objects for import. 
* Removed corresponding globals referring to their index val in the MAF header.
* Removed corresponding references to `MafRecord` setters

**Comment by angelicaochoa** [comment]
@n1zea144 @pieterlukasse 
Can you comment on the below? Let me know if you have any questions.

**Open up discussion for what to do with MAF fields which _could_ map directly to database fields but are ignored during import and/or not used to set corresponding fields when constructing database objects for import (`mutation`, `mutation_event`)**
```
- ONCOTATOR_REFSEQ_MRNA_ID (`"ONCOTATOR_REFSEQ_MRNA_ID"`) --> `mutation_event.oncotator_refseq_mrna_id` (importer only looks at "RefSeq")
- ONCOTATOR_UNIPROT_ENTRY_NAME (`"ONCOTATOR_UNIPROT_ENTRY_NAME"`) --> `mutation_event.oncotator_uniprot_entry_name` (this field is never set)
- ONCOTATOR_UNIPROT_ACCESSION (`"ONCOTATOR_UNIPROT_ACCESSION"`) --> `mutation_event.oncotator_uniprot_accession` (importer only looks at "SWISSPROT")
- ONCOTATOR_CODON_CHANGE (`"ONCOTATOR_CODON_CHANGE"`) --> `mutation_event.oncotator_codon_change`(importer only looks at "Codons")
- ONCOTATOR_PROTEIN_POS_START (`"ONCOTATOR_PROTEIN_POS_START"`) --> `mutation_event.oncotator_protein_pos_start` (importer resolves from record.getProteinChange(), record.getProteinPosition() and ignores this Oncotator field)
- ONCOTATOR_PROTEIN_POS_END (`"ONCOTATOR_PROTEIN_POS_END"`) --> `mutation_event.oncotator_protein_pos_end` (same as ONCOTATOR_POS_START)
```

File: core/src/main/java/org/mskcc/cbio/maf/MafUtil.java [hunk 5]
```diff
@@ -208,79 +180,57 @@
...
-        {
+        for (int i=0; i<parts.length; i++) {
             String header = parts[i];
 
-	        // put the index to the map
-	        this.columnIndexMap.put(header.toLowerCase(), i);
+            // put the index to the map
+            this.columnIndexMap.put(header.toLowerCase(), i);
 
-	        // determine standard & default column indices
```
Comment (sheridancbio) [inline core/src/main/java/org/mskcc/cbio/maf/MafUtil.java]: side comment : This has runtime order(n^2) on the number of columns in the maf that we recognize. There is a better way to scan the column headers and determine the indexes. But even with 100 columns, it is only ~ 10000 string comparisons which will only waste something like 1 or 2 seconds per MAF

File: core/src/main/java/org/mskcc/cbio/portal/util/ExtendedMutationUtil.java (hunk 1 above)
Comment (sheridancbio) [reply core/src/main/java/org/mskcc/cbio/portal/util/ExtendedMutationUtil.java]: this indicates that for this field at least, it was a conscious decision to move away from the oncotator value

**Comment by n1zea144** [comment]
> **Open up discussion for what to do with MAF fields which _could_ map directly to database fields but are ignored during import and/or not used to set corresponding fields when constructing database objects for import (`mutation`, `mutation_event`)**


@angelicaochoa This sounds like code that was put in place when we were using Oncotator for annotation, but has become obsolete when we moved to VEP annotation.  I think we should remove these fields since they are no longer used and only add to confusion to an area of the codebase that is hard to follow to begin with.

As an added housekeeping suggestion, we should remove any  ONCOTATOR_* fields in the database schema that are not used by the code and rename the remaining ones (at least chop off the ONCOTATOR_ suffix).

I've added @onursumer to the conversation as he may provide some useful historical context to this part of the codebase.

**Comment by angelicaochoa** [comment]
@onursumer can you comment on this?

**Comment by onursumer** [comment]
Agreed with @n1zea144. Ideally we should not have anything related to `oncotator` neither in the database nor in the code base.

I am not even sure at this point if the MAF parsing logic should still stay in the main `cbioportal` repo. We should probably have a separate repo where we process files and import them to the database, but I think that's beyond the scope of this PR.

**Comment by jjgao** [comment]
@n1zea144 @ao508 what's the plan on this?

**Comment by ao508** [comment]
@n1zea144 rebased

File: docs/Imported-MAF-Columns.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
+Relevant classes:
+* [MafUtil](https://github.com/cBioPortal/cbioportal/blob/master/core/src/main/java/org/mskcc/cbio/maf/MafUtil.java)
+* [ExtendedMutationutil](https://github.com/cBioPortal/cbioportal/blob/master/core/src/main/java/org/mskcc/cbio/portal/util/ExtendedMutationUtil.java)
```
Comment (mandawilson) [inline docs/Imported-MAF-Columns.md]: The "u" in "util" should be capitalized.
